### PR TITLE
Start a new pipeline build with resource parameters

### PR DESCRIPTION
**Pull request by MilenaHristova**
https://github.com/dotnet/arcade-services/issues/2925

### Release Note Category
- [ ] Feature changes/additions 
- [ ] Bug fixes
- [x] Internal Infrastructure Improvements

### Release Note Description
Add pipeline resource parameters to StartNewBuildAsync method

### CODE REVIEW

File: src/Microsoft.DotNet.Darc/DarcLib.AzDev/AzureDevOpsClient.cs [hunk 1]
```diff
@@ -1179,19 +1179,35 @@ public async Task<int> StartNewReleaseAsync(string accountName, string projectNa
...
-        foreach ((string name, string value) in queueTimeVariables)
+
+        if (queueTimeVariables != null)
         {
-            variables.Add(name, new AzureDevOpsVariable(value));
+            foreach ((string name, string value) in queueTimeVariables)
+            {
+                variables.Add(name, new AzureDevOpsVariable(value));
+            }
+        }
```
Comment (premun) [inline src/Microsoft.DotNet.Darc/DarcLib.AzDev/AzureDevOpsClient.cs:1201]: Not sure if you'd prefer that but you could do:

```csharp
var variables = queueTimeVariables?.ToDictionary<>(...) ?? new Dictionary<string, AzureDevOpsVariable>();
```

Same below

Comment (MilenaHristova) [reply src/Microsoft.DotNet.Darc/DarcLib.AzDev/AzureDevOpsClient.cs:1201]: but queueTimeVariables is a dictionary - does it work like that?

Comment (dkurepa) [reply src/Microsoft.DotNet.Darc/DarcLib.AzDev/AzureDevOpsClient.cs:1201]: It should, you're basically doing the same thing with the `foreach`

Comment (premun) [reply src/Microsoft.DotNet.Darc/DarcLib.AzDev/AzureDevOpsClient.cs:1201]: `Dictionary is also `IEnumerable<KeyValuePair<K, V>>`

Comment (MilenaHristova) [reply src/Microsoft.DotNet.Darc/DarcLib.AzDev/AzureDevOpsClient.cs]: took some time but I figured it out

**Review by premun — DISMISSED** [review]

File: src/Microsoft.DotNet.Darc/DarcLib.AzDev/AzureDevOpsClient.cs [hunk 2]
```diff
@@ -1179,20 +1179,24 @@ public async Task<int> StartNewReleaseAsync(string accountName, string projectNa
     /// <param name="azdoDefinitionId">ID of the build definition where a build should be queued.</param>
     /// <param name="queueTimeVariables">Queue time variables as a Dictionary of (variable name, value).</param>
     /// <param name="templateParameters">Template parameters as a Dictionary of (variable name, value).</param>
+    /// /// <param name="pipelineResources">Pipeline resources as a Dictionary of (pipeline resource name, build number).</param>
```
Comment (dkurepa) [inline src/Microsoft.DotNet.Darc/DarcLib.AzDev/AzureDevOpsClient.cs:1182]: nit: remove the extra ///

Comment (MilenaHristova) [reply src/Microsoft.DotNet.Darc/DarcLib.AzDev/AzureDevOpsClient.cs]: done

**Review by dkurepa — APPROVED** [review]

**Pull request merged**